### PR TITLE
Added FollowPathBehaviour, TeleportBehavior, and a Trail

### PR DESCRIPTION
**Pull request by steffanmouton**

### CODE REVIEW

File: Assets/Scripts/Steffan/TeleportBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+public class TeleportBehaviour : MonoBehaviour {
+	[Header("This is the target destination for the teleporting object.")]
+	public Transform targetPosition;
+	[Header("This is the object that will be teleported to the destination.")]
+	public Transform sourceObject;
+
+	public UnityEvent response;
+	// Use this for initialization
+	void Start () {
```
Comment (chuwilliamson) [inline Assets/Scripts/Steffan/TeleportBehaviour.cs]: remove unused functions

File: Assets/Scripts/Steffan/TeleportBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		
+	}
+	
+	// Update is called once per frame
+	void Update () {
```
Comment (chuwilliamson) [inline Assets/Scripts/Steffan/TeleportBehaviour.cs]: move this into an InputBehaviour class that will be given the references

File: Assets/Scripts/Steffan/FollowPathBehaviour.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	private void Start()
+	{
+		movement = MoveMethod.TRAVEL;
+		SavedPath = new List<GameObject>(WaypointsToFollow);
+	}
+
+	void RestartPath()
+	{
+		WaypointsToFollow = new List<GameObject>(SavedPath);
+		transform.position = WaypointsToFollow[0].transform.position;
```
Comment (chuwilliamson) [inline Assets/Scripts/Steffan/FollowPathBehaviour.cs:46]: use a list of transforms in place of a list of gameObjects

File: Assets/Scripts/Steffan/FollowPathBehaviour.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+	
+	[SerializeField]
+	private void AddWaypointPassed(GameObject go)
+	{
+		WaypointsPassed.Insert(0, go);
+	}
+	
+	private void OnTriggerEnter(Collider other)
+	{
+		if (!other.gameObject.CompareTag("Waypoint"))
```
Comment (chuwilliamson) [inline Assets/Scripts/Steffan/FollowPathBehaviour.cs:84]: make a string field with the Cinemachine.TagField attribute

**Review by chuwilliamson — CHANGES_REQUESTED** [review]
Changes should result in an approved request

File: Assets/Scripts/Steffan/FollowPathBehaviour.cs (hunk 4 above)
Comment (steffanmouton) [reply Assets/Scripts/Steffan/FollowPathBehaviour.cs:84]: Idk what this does, but yessir.

**Comment by steffanmouton** [comment]
Requested Changes addressed
Added Cinemachine to project

**Review by chuwilliamson — APPROVED** [review]

**Pull request merged**